src/main/antlr/Grammar.g4

```diff
@@ -1,41 +1,84 @@
-
 grammar Grammar;
 
+simpleEq
+    : (left=var EQ)? right=expr SEMICOLON? EOF
+    ;
+
+var
+    : IDENTIFIER
+    ;
+
 expr
     : '(' expr ')'                                     #default
-    | left=expr operator=(PLUS | MINUS) right=expr     #expression
-    | value                                            #defaultValue
-    | func '(' expr ')'                                #defaultFunc
+    | left=expr op=operator right=expr                 #defaultOperation
+    | numericValue                                     #defaultValue
+    | euler                                            #defaultEuler
+    | name=func '(' expr ')'                           #defaultFunc
     ;
 
 func
     : SIN
     | COS
+    | TAN
+    ;
+
+numericValue
+    : integerValue
+    | decimalValue
+    ;
+
+integerValue
+    : INT_VALUE
+    ;
+
+decimalValue
+    : DECIMAL_VALUE
     ;
 
-value
-    : INT
-//    | DECIMAL
+euler
+    : 'e'
     ;
 
+operator
+    : PLUS
+    | MINUS
+    | TIMES
+    | DIVIDED
+    ;
+
+EQ: '=';
+
 SIN: 'sin';
 COS: 'cos';
+TAN: 'tan';
+
 PLUS: '+';
 MINUS: '-';
+TIMES: '*';
+DIVIDED: '/';
+
 SEMICOLON: ';';
 
+fragment LETTER
+    : [a-zA-Z]
+    ;
+
 fragment DIGIT
     : [0-9]
     ;
 
-INT
+IDENTIFIER
+    : (LETTER | '_') (LETTER | DIGIT | '_' | '@')*
+    ;
+
+INT_VALUE
     : DIGIT+
     ;
 
-//DECIMAL
-//    : DIGIT+ ('.' (DIGIT+)?)
-//    | ('.' DIGIT+)
-//    ;
+DECIMAL_VALUE
+    : DIGIT+ ('.' DIGIT+)?
```

Comment:
`?` will never be met, because integerValue in numericValue will match it first.